subsys/net/lib/shell/qbv_shell.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include <zephyr/shell/shell.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/ethernet_mgmt.h>
+
+#include "net_shell_private.h"
+
+/* qbv enable <iface_index> <value(off, on)> */
+static int cmd_net_qbv(const struct shell *sh, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+#if defined(CONFIG_NET_QBV)
+	shell_print(sh, "To set Qbv config:");
+	shell_print(sh, "  1. Run enable to on");
+	shell_print(sh, "  2. Run set_config to set base_time/cycle_time/cycle_time_ext/list_len");
+	shell_print(sh, "  3. Run set_gc to set gate control");
+	shell_print(sh, "For example:");
+	shell_print(sh, "  1. net qbv enable 1 on");
+	shell_print(sh, "  2. net qbv set_config 1 200 0 0 10000000 0 2");
+	shell_print(sh, "  3. qbv set_gc 1 0 0x1 5000000");
+	shell_print(sh, "  4. qbv set_gc 1 0 0x2 5000000");
+#else
+	shell_print(sh, "Set %s to enable %s support.\n", "CONFIG_NET_QBV", "qbv");
+#endif
+
+	return 0;
+}
+
+/* qbv enable <iface_index> <value(off, on)> */
+static int cmd_qbv_enable(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_if *iface;
+	struct ethernet_req_params params;
+	int ret;
+	bool enable;
+	int idx;
+
+	idx = get_iface_idx(sh, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
```

Comment:
What is missing here is verification that the interface is Ethernet type. That can be done like this
```
	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
		PR_WARNING("Qbv can be set only for Ethernet\n");
		return -ENOEXEC;
	}
```